### PR TITLE
Request ms update for pending files

### PR DESCRIPTION
**Pull request by andrecardoso**
This is needed to fix https://github.com/Rise-Vision/rise-launcher-electron/issues/745

### CODE REVIEW

**Comment by andrecardoso** [comment]
[(2) Defect 745: Local storage does not send NOEXIST message for trashed/deleted files](https://trello.com/c/Iptu0LyA/4469-2-defect-745-local-storage-does-not-send-noexist-message-for-trashed-deleted-files)

File: src/messaging/watch/watch.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ function processFileWatch(message, existingMetadata) {
   const metadata = existingMetadata || {filePath: message.filePath};
   metadata.status = metadata.status || "UNKNOWN";
 
-  if (metadata.status === "UNKNOWN") {
+  if (metadata.status === "UNKNOWN" || metadata.status === "PENDING") {
```
Comment (rudibravo) [inline src/messaging/watch/watch.js]: @andrecardoso I'm not sure I follow why the pending status should also request an update from MS. Doesn't it mean it's already waiting for such update?

Comment (andrecardoso) [reply src/messaging/watch/watch.js]: I'm adding this to address the following scenario:
1 - first WATCH is received 
2 - entry is marked as PENDING in the database
3 - player is closed before the response arrives or the response is an error or any other situation where the status is not updated in the database
4 - player is launched again, status is PENDING, local storage does nothing

Comment (andrecardoso) [reply src/messaging/watch/watch.js]: Basically this is to request updates for previous PENDING entries

Comment (andrecardoso) [reply src/messaging/watch/watch.js]: Alternatively we could remove the PENDING state completely as it is not being read anywhere besides unit tests. I don't know the exact reason we have it. @tejohnso what do you think?

Comment (tejohnso) [reply src/messaging/watch/watch.js]: I don't see a purpose for it either. It wasn't part of the original design. @stulees do you see any reason for it?

Comment (stulees) [reply src/messaging/watch/watch.js]: I think it was to indicate the file is pending a response from MS, but the client side doesn't need to know, and I'm sure enough has changed in LS now that this isn't relevant/required anymore.

Comment (tejohnso) [reply src/messaging/watch/watch.js]: Thanks @stulees 
@andrecardoso I think we should go ahead and scrap it. If the need presents itself in the future we can reassess then.

Comment (andrecardoso) [reply src/messaging/watch/watch.js]: Thanks @stulees @tejohnso

Comment (andrecardoso) [reply src/messaging/watch/watch.js]: I'm going to close this PR and open another removing the PENDING state

**Comment by rudibravo** [comment]
@andrecardoso I think we can delete this branch right?

**Comment by andrecardoso** [comment]
@rudibravo correct. Deleted! thanks

**Pull request closed without merging**